src/werkzeug/sansio/request.py

```diff
@@ -179,6 +180,62 @@ def is_secure(self) -> bool:
         "`True` if the request is secure."
         return self.scheme in {"https", "wss"}
 
+    @cached_property
+    def url(self) -> str:
+        """The reconstructed current URL as IRI.
+        See also: :attr:`trusted_hosts`.
+        """
+        return get_current_url(
+            self.scheme,
+            self.host,
+            self.path,
+            self.root_path,
+            self.query_string,
+        )
+
+    @cached_property
+    def base_url(self) -> str:
+        """Like :attr:`url` but without the querystring
+        See also: :attr:`trusted_hosts`.
+        """
+        return get_current_url(
+            self.scheme,
+            self.host,
+            self.path,
+            self.root_path,
+            self.query_string,
+            strip_querystring=True,
+        )
+
+    @cached_property
+    def url_root(self) -> str:
```

Comment:
This likely doesn't match the pattern of the other `url_` names to match `script_root`. I'll rename this `root_url` and add `url_root` as an alias next to the `script_root` alias in `wrappers.Request`.